# Multiple-Sclerosis-RNA-Seq
See Wiki section for information about the project.

# Functionality 
The initial input files were provided by Dr. Chen and made accessible to us on our class server. Each experimental group has three FASTQ files of RNA Seq transcripts.  Kallisto quantification is run using the generated index and the transcript FASTQ files(**See Kallisto script Project_6.py**). After that, the three output files produced by Kallisto for each of the experimental groups are read in by Sleuth, and the comparisons: CFA vs. Vehicle, CFA vs. Sephin1, and Vehicle vs. Sephin1 between the groups are done. Within Sleuth, the Wald test function is included to compute the specific beta coefficient of every transcript, which is necessary for WebGestalt(**See Sleuth code Sleuth_WaldTest.R**). Furthermore, the transcript IDs are converted to gene IDs using Biomart. Next, three tables (for the three different comparisons) are generated with two columns for gene IDs and their beta coefficients. These tables are uploaded separately to the web-based WebGestalt for gene set enrichment analysis (GSEA). GSEA allows for the extraction of biological insight from genes of interest and generates volcano plots and other figures that aid in visualizing their changed pathways. The final output files of our pipeline includes a gene list and the plots generated by WebGestalt. 

- Input 
  - Single-ended FASTQ files of RNA Seq transcripts 
- Outputs 
  - Kallisto files
  - Ranked gene list
  - Plots generated by WebGestalt
- Environments
  - Kallisto script --> Class Server/Python
  - Sleuth code --> RStudio

# Dependancies
Package installation for Python:
```
import os 
import Bio
from Bio import Entrez
```

Package installation for R:
```
library(sleuth)
library(dplyr)

source("https://bioconductor.org/biocLite.R")
biocLite("biomaRt")

library('cowplot')
library('sleuth')
```
Programs:

- Kallisto
  - Command line

- Sleuth
  - R Library

Web-Based tools:

- WebGestalt




